sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1039,6 +1039,83 @@ case class GpuArraysOverlap(left: Expression, right: Expression)
   }
 }
 
+case class GpuArrayRemove(left: Expression, right: Expression) extends GpuBinaryExpression {
+
+  override def dataType: DataType = left.dataType
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    // For every null entry in rhs, replace corresponding row in lhs with null
+    val lhsWithNull = withResource(rhs.getBase.isNull) { rhsIsNull =>
+      withResource(GpuScalar.from(null, dataType)) { nullList =>
+        rhsIsNull.ifElse(nullList, lhs.getBase)
+      }
+    }
+    // Repeat every entry in rhs n times where n is number of elements in corresponding row in lhs
```

Comment:
This approach seems to be very inefficient. Instead of materializing the search keys and compare with the search space, we can just search the indices of the keys in the search space, then gather. That requires a new `lists::index_of` implementation from cudf, but should be much better than this approach. Imagine that the search space (lists) has very large size and can be complex nested type, materializing the keys (rhs) can incur a lot of memory and computation overhead.

Reply:
I've filed a cudf issue for it: https://github.com/rapidsai/cudf/issues/12212.

Reply:
Note that I'm totally fine if we need to prioritize this to completion (keep the current implementation as-is) and then do the follow-up work (with new cudf API) for optimizing performance later.

Reply:
To be clear the common case here is with RHS being a scalar value. I would like to see a specialized version for RHS as a scalar which would make all of this much simpler.

Reply:
@ttnghia Thank you very much! I will follow up with the cudf issues above.